resources/logfs/header.html

```diff
@@ -1,7 +1,148 @@
-                                                                                                   
-                                                                                                   
-        
-<!--FS_START
\ No newline at end of file
+<html>
+  <head>
+    <meta charset="utf-8" />
+    <style>
+      /* Horizontal stack */
+      .bb {
+        display: flex;
+      }
+      .bb > * + * {
+        margin-left: 10px;
+      }
+      /* Button further styled by dl.css using p and s classes */
+      button {
+        display: block;
+      }
+      body {
+        font-family: sans-serif;
+        margin: 1em;
+      }
+      table {
+        border-collapse: collapse;
+        width: 50%;
+        margin-top: 1em;
+      }
+      tr:first-child {
+        font-weight: bold;
+      }
+      tr:not(:first-child) {
+        text-align: right;
+      }
+      td {
+        border: 1px solid #ddd;
+        padding: 8px;
+        min-width: 8ch;
+      }
+      iframe {
+        display: none;
+      }
+    </style>
+    <link rel="stylesheet" href="https://microbit.org/dl/1/dl.css" />
+    <script>
+      // These help significantly to save bytes.
+      let w = window;
+      let d = document;
+      let l = w.location;
+      let n = null;
+      let csv = "";
+      let tag = document.createElement.bind(document);
+      // Allow external JS to override these actions if we managed to load it.
+      // This script block is first so external JS can override individual functions.
+      w.dl = {
+        download: function () {
+          let a = tag("a");
+          a.download = "microbit.csv";
+          a.href = URL.createObjectURL(new Blob([csv], { type: "text/plain" }));
+          a.click();
+          a.remove();
+        },
+        copy: function () {
+          navigator.clipboard.writeText(csv.replace(/\,/g, "\t"));
+        },
+        update: alert.bind(
+          n,
+          "Unplug your micro:bit, then plug it back in and wait"
+        ),
+        clear: alert.bind(
+          n,
+          "The log is cleared when you reflash your micro:bit"
+        ),
+        load: function () {
+          // Find the CSV text in the file system
+          let p = d.querySelector("#w");
+          let raw = d.documentElement.outerHTML.split("FS_START")[2];
+          // See MicroBitLog.h/cpp for the format.
+          if (/^UBIT_LOG_FS_V_001/.test(raw)) {
+            let dataStart = parseInt(raw.substr(29, 10), 16) - 2048;
+            let dataSize = 0;
+            while (raw.charCodeAt(dataStart + dataSize) != 0xfffd) {
+              dataSize++;
+            }
```

Comment:
Does your minifier take out the superfluous braces here?

Reply:
Yep. It ends up as a for loop with no braces.

Reply:
nice!